File: Final/erp-backend/src/main/java/com/pyme/erp/service/VentaService.java
package com.pyme.erp.service;

import com.pyme.erp.dto.request.VentaRequest;
import com.pyme.erp.model.Venta;

import java.time.LocalDateTime;
import java.util.List;

public interface VentaService {
    
    Venta crear(VentaRequest request);
    
    Venta obtenerPorId(Long id);
    
    List<Venta> listarTodas();
    
    List<Venta> obtenerPorCliente(Long clienteId);
    
    List<Venta> obtenerPorEstado(Venta.EstadoVenta estado);
    
    List<Venta> obtenerPorRangoFechas(LocalDateTime fechaInicio, LocalDateTime fechaFin);
    
    Venta cambiarEstado(Long id, Venta.EstadoVenta nuevoEstado);
    
    void cancelar(Long id);
}

File: Final/erp-frontend/src/app/services/venta.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ResponseWrapper, Venta, EstadoVenta } from '../models/response-wrapper.model';

@Injectable({
  providedIn: 'root'
})
export class VentaService {
  private baseUrl = 'http://localhost:8080/api/ventas';

  constructor(private http: HttpClient) { }

  crear(venta: any): Observable<ResponseWrapper<Venta>> {
    return this.http.post<ResponseWrapper<Venta>>(this.baseUrl, venta);
  }

  listar(): Observable<ResponseWrapper<Venta[]>> {
    return this.http.get<ResponseWrapper<Venta[]>>(this.baseUrl);
  }

  obtenerPorId(id: number): Observable<ResponseWrapper<Venta>> {
    return this.http.get<ResponseWrapper<Venta>>(`${this.baseUrl}/${id}`);
  }

  obtenerPorCliente(clienteId: number): Observable<ResponseWrapper<Venta[]>> {
    return this.http.get<ResponseWrapper<Venta[]>>(`${this.baseUrl}/cliente/${clienteId}`);
  }

  obtenerPorEstado(estado: EstadoVenta): Observable<ResponseWrapper<Venta[]>> {
    return this.http.get<ResponseWrapper<Venta[]>>(`${this.baseUrl}/estado/${estado}`);
  }

  cambiarEstado(id: number, estado: EstadoVenta): Observable<ResponseWrapper<Venta>> {
    return this.http.put<ResponseWrapper<Venta>>(`${this.baseUrl}/${id}/estado?estado=${estado}`, {});
  }

  cancelar(id: number): Observable<ResponseWrapper<void>> {
    return this.http.delete<ResponseWrapper<void>>(`${this.baseUrl}/${id}`);
  }
}